packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/update.ts

```diff
@@ -32,84 +34,144 @@ export interface PerformUpdateParams<T = unknown> {
 }
 
 export const performUpdate = async <T>(
+  updateParams: PerformUpdateParams<T>,
+  apiContext: ApiExecutionContext
+): Promise<SavedObjectsUpdateResponse<T>> => {
+  const { type, id, options } = updateParams;
+  const { allowedTypes, helpers } = apiContext;
+  const namespace = helpers.common.getCurrentNamespace(options.namespace);
+
+  // check request is valid
+  const { validRequest, error } = isValidRequest({ allowedTypes, type, id });
+  if (!validRequest && error) {
+    throw error;
+  }
+
+  const maxAttempts = options.version ? 1 : 1 + DEFAULT_RETRY_COUNT;
+
+  // handle retryOnConflict manually by reattempting the operation in case of conflict errors
+  let response: SavedObjectsUpdateResponse<T>;
+  for (let currentAttempt = 1; currentAttempt <= maxAttempts; currentAttempt++) {
+    try {
+      response = await executeUpdate(updateParams, apiContext, { namespace });
+      break;
+    } catch (e) {
+      if (
+        SavedObjectsErrorHelpers.isConflictError(e) &&
+        e.retryableConflict &&
+        currentAttempt < maxAttempts
+      ) {
+        continue;
+      }
+      throw e;
+    }
+  }
+
+  return response!;
+};
+
+export const executeUpdate = async <T>(
   { id, type, attributes, options }: PerformUpdateParams<T>,
-  {
-    registry,
-    helpers,
-    allowedTypes,
-    client,
-    serializer,
-    migrator,
-    extensions = {},
-  }: ApiExecutionContext
+  { registry, helpers, client, serializer, extensions = {}, logger }: ApiExecutionContext,
+  { namespace }: { namespace: string | undefined }
```

Comment:
The actual update workflow. The comments should in theory be sufficient to understand the flow